### DISCUSSION TITLE
Transactions support for Elasticsearch but not Postgres provider, rationale?

### DISCUSSION THREAD

**Discussion by staffordsmith83** [geopython/pygeoapi#1254] [Q&A] [answered]
Hi, thanks so much for supporting transactions! I see of all the providers, only Elasticsearch can be configured to be editable, as per [this matrix here](https://docs.pygeoapi.io/en/latest/data-publishing/ogcapi-features.html#providers).

I always use Postgres/PostGIS for my geospatial database due to the powerful spatial functions. Im interested in the reason why Elasticsearch has been chosen to be the only provider that supports transactions.

Is Elasticsearch emerging as a powerful alternative to PostGIS?
Is it planned to enable transaction support in the Postgres provider?
Should I consider switching to use Elasticsearch as my goto spatial data store instead?

thanks again for your development efforts!

**Comment by tomkralidis** [comment, marked as answer]
The basic principle of pygeoapi providers/plugins is that they are maintained as required.  This means that as a feature is added in a given plugin, the goal is not to maintain parity across all plugins.  This allows plugins to evolve independently and that a provider contributor is not on the hook to implement across all providers.
